bluetoe/link_layer/include/bluetoe/peripheral_latency.hpp

```diff
@@ -0,0 +1,318 @@
+#ifndef BLUETOE_LINK_LAYER_PERIPHERAL_LATENCY_HPP
+#define BLUETOE_LINK_LAYER_PERIPHERAL_LATENCY_HPP
+
+#include <bluetoe/ll_meta_types.hpp>
+
+/**
+ * @file bluetoe/peripheral_latency.hpp
+ *
+ * This file provides several options, how Bluetoe will implement peripheral latency. Peripheral latency
+ * allows a link layer peripheral to miss some connection events (do not respond to the central) to
+ * conserve power.
+ *
+ * Each option is a compromise between latency and power consumption.
+ *
+ * @sa bluetoe::link_layer::peripheral_latency_ignored
+ * @sa ...
+ */
+namespace bluetoe {
+namespace link_layer {
+
+    namespace details {
+        struct peripheral_latency_meta_type {};
+    }
+
+    /**
+     * @brief detailed options for the peripheral latency behavior
+     *
+     * Every option defines a set of events / circumstances under which
+     * the link layer should listen to the very next connection event.
+     * (beside the minumum events, the link layer has to listen to due
+     * to the periperhal latency value of the connection).
+     *
+     * @note in combination, there are some options that make no sense.
+     *       For example, using listen_always with some of the other
+     *       listen_* options is pointless. Also combining send_always
+     *       with send_pending_data doesn not make sense.
+     */
+    enum class peripheral_latency
+    {
+        /**
+         * @brief listen at all connection events.
+         *
+         * This does not imply, that the link layer will also respond at every
+         * connection event.
+         */
+        listen_always,
```

Comment:
I'm not sure is clear from what I've read that the slave latency negotiated for the connection does not change. And that all these options are variations on how the peripheral will behave within the agreed connection parameters.

Reply:
How about:

         * @brief listen at all connection events, despite the current
         *        peripheral latency value of the current connection.
         *
         * This does not imply, that the link layer will also respond at every
         * connection event. Just listening, without responding will
         * synchronize the link layer with the central and thus narrow the
         * receive window size at the next connection interval.

Reply:
That looks good. Though it would still seem to miss the idea that (maybe obvious for us) the negotiated peripheral latency for the connection does not change.